demo/Views/WelcomeView.vala

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2017 elementary LLC. (https://elementary.io)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+public class WelcomeView : Gtk.Grid {
+    construct {
+        var welcome = new Granite.Widgets.Welcome ("Granite Demo", "This is a demo of the Granite library.");
+        welcome.append ("text-x-vala", "Visit Valadoc", "The canonical source for Vala API references.");
+        welcome.append ("text-x-source", "Get Granite Source", "Granite's source code is hosted on GitHub.");
+
+        add (welcome);
+
+        welcome.activated.connect ((index) => {
+            switch (index) {
+                case 0:
+                    try {
+                        AppInfo.launch_default_for_uri ("https://valadoc.org/granite/Granite.html", null);
+                    } catch (Error e) {
+                        warning ("%s\n", e.message);
```

Comment:
You don't need this "%s\n" string formatting here, `warning ()` takes care of that, just do:
`warning (e.message)`.